.github/workflows/e2e_nightly_upgrade.yml

```diff
@@ -0,0 +1,204 @@
+          bash -c 'while [[ "$(curl -L -s -o /dev/null -w %{http_code} http://localhost:8001/index.php)" != "200" ]]; do sleep 5; done'
+
+      # Install Module with user interface
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 10
+
+      - name: Install dependencies
+        run: npm install
+        working-directory: ./tests/e2e
+
+      - name: Install autoupgrade module
+        run: npm run install-module
+        working-directory: ./tests/e2e
+        env:
+          URL_FO: "http://localhost:8001/"
+          PS_VERSION: ${{ matrix.ps_start_version }}
+
+      - name: Download nightly build
+        run: |
+          docker exec -u www-data prestashop_autoupgrade mkdir -p admin-dev/autoupgrade/download
+          docker exec -u www-data prestashop_autoupgrade curl ${{ matrix.ps_target_version.archive_zip }} -o admin-dev/autoupgrade/download/${{ matrix.ps_target_version.filename }}
+          docker exec -u root prestashop_autoupgrade chmod 777 -R /var/www/html
+
+      # Run Upgrade tests
+      - name: Run tests
+        run: npm run upgrade-test
+        working-directory: ./tests/e2e
+        env:
+          URL_FO: "http://localhost:8001/"
+          PS_VERSION: ${{ matrix.ps_start_version }}
+          PS_VERSION_UPGRADE_TO: ${{ matrix.ps_target_version.for_test }}
+          ZIP_NAME: ${{ matrix.ps_target_version.filename }}
+
+      - uses: actions/upload-artifact@v2
+        name: Upload failed screenshots
+        if: ${{ always() }}
+        with:
+          name: failed-steps-for-test-${{ matrix.ps_start_version }}-${{ matrix.ps_target_version.for_test }}
+          path: ./tests/e2e/screenshots/
+
+      # Upload Mochawesome reports
+      - name: Rename Mochawesome report
+        if: ${{ always() }}
+        run: |
+          mkdir -p ./json-reports
+          cp ./tests/e2e/mochawesome-report/mochawesome.json ./json-reports/upgrade-from-${{ matrix.ps_start_version }}-to-${{ matrix.ps_target_version.for_test }}.json
+
+      - uses: actions/upload-artifact@v2
+        name: Upload Mochawesome report
+        if: ${{ always() }}
+        with:
+          name: tests-reports
+          path: ./json-reports
+
+  # Second job: merge reports (depends on upgrade job)
+  merge-reports:
+    name: Combine all tests reports
+    needs: upgrade
+    runs-on: ubuntu-18.04
+
+    env:
+      report_path: ./json-reports
+      combined_report_name: upgrade_tests.json
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/download-artifact@v2
+        name: Download reports
+        with:
+          name: tests-reports
+          path: ${{ env.report_path }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Combine reports
+        shell: bash
+        run: ./tests/e2e/scripts/combine-reports.py "${{ env.report_path }}" "${{ env.report_path }}/${{ env.combined_report_name }}"
```

Comment:
Same about combine task, it's a little bit complicated and can be improved by using dedicated scripts

Reply:
done